LoRaEngine/LoraKeysManagerFacade/IoTCentralImp/IoTCentralDeviceRegistryManager.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade.IoTCentralImp
+{
+    using System;
+    using System.Globalization;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Formatting;
+    using System.Threading.Tasks;
+    using LoraKeysManagerFacade.IoTCentralImp.Definitions;
+    using Newtonsoft.Json.Serialization;
+
+    internal class IoTCentralDeviceRegistryManager : IDeviceRegistryManager
+    {
+        private const string API_VERSION = "api-version=1.0";
+
+        private readonly IDeviceProvisioningHelper provisioningHelper;
+        private readonly HttpClient client;
+        private readonly JsonMediaTypeFormatter formatter;
+
+        public IoTCentralDeviceRegistryManager(HttpClient client, IDeviceProvisioningHelper provisioningHelper)
+        {
+            this.client = client;
+            this.provisioningHelper = provisioningHelper;
+            this.formatter = new JsonMediaTypeFormatter
+            {
+                SerializerSettings = new Newtonsoft.Json.JsonSerializerSettings
+                {
+                    ContractResolver = new CamelCasePropertyNamesContractResolver()
+                }
+            };
+        }
+
+        public async Task<IDevice> GetDeviceAsync(string deviceId)
+        {
+            var deviceRequest = await this.client.GetAsync(new Uri(this.client.BaseAddress, $"/api/devices/{deviceId}?{API_VERSION}"));
+
+            if (deviceRequest.StatusCode == HttpStatusCode.NotFound)
+            {
+                return null;
+            }
+
+            var deviceResponse = await deviceRequest.Content.ReadAsAsync<Device>();
+
+            var attestation = this.provisioningHelper.ProvisionDevice(deviceId, out var assignedIoTHubHostname);
+
+            return new IoTCentralDevice(deviceResponse, attestation, assignedIoTHubHostname);
```

Comment:
If attestation is null (device could not be provisioned) you probably want to return null here as well

Reply:
I implemented your suggestion